third_party/move/move-compiler-v2/tests/checking/typing/lambda.exp

```diff
@@ -6,7 +6,7 @@ error: `M::reduce` is a function and not a macro
 34 │         foreach(&v, |e| sum = sum + reduce!(*e, 0, |t, r| t + r));
    │                                     ^^^^^^
 
-error: invalid call of `M::foreach`: expected `&?19` but found `integer` for argument 2
+error: invalid call of `M::foreach`: expected `integer` but found `&?19` for argument 2
```

Comment:
This error message is wrong, as we shouldn't expect `integer` but `function returning integer` for argument 2.  As this is likely this is beyond the scope of the current change, do we have an issue to track this problem?  If not, open an issue, please.

Reply:
It is not wrong, it is contra-variance and that is how it looks like for now. We can come back later, but it is logically correct.

Reply:
This error message is wrong.  I've filed a new issue for it since you don't seem to be listening here.

https://github.com/aptos-labs/aptos-core/issues/11055